### PR TITLE
added UITextField to RxAlert

### PR DESCRIPTION
**Pull request by keisukeYamagishi**
# Abstract

UITextField was added because RxAlert did not implement UITextField.
You can use multiple UITextFields.
You can customize UITextField with RxAlert.
You can use the same interface as before.
So you can use RxAlert in the same way as before.

## added function

You can use UITextField with RxAlert
You can customaize UITextField with RxAlert
added default parameter at AlertAction

## issues

#36

### CODE REVIEW

File: RxAlert/UIViewController+.swift [hunk 1]
```diff
@@ -28,15 +28,21 @@ import RxCocoa
...
-                })
-            }.forEach(self.addAction)
+            actions.forEach { action in
+                if let textField = action.textField {
+                    alert.addTextField { text in
+                        text.config(textField)
+                    }
+                }else{
+                    self.addAction(UIAlertAction(title: action.title, style: action.style, handler: { _ in
+                        observer.on(.next((OutputAction(index: action.type, textFields: alert.textFields))))
```
Comment (susan335) [inline RxAlert/UIViewController+.swift]: Is there a memory leak?
`OutputAction(index: action.type, textFields: alert.textFields)`
`alert` is `self`.

How about this...
```
handler: { [weak self]_ in
                         observer.on(.next((OutputAction(index: action.type, textFields: self?.textFields))))
```

Maybe the second parameter is unnecessary
`func addAction(actions: [AlertAction], 👉 alert: UIAlertController)`

Comment (keisukeYamagishi) [reply RxAlert/UIViewController+.swift]: `alert parameter` was not need so I removed this parameter but memory leak is not occur so it is not need at the `[weak self]`

Comment (susan335) [reply RxAlert/UIViewController+.swift]: Sorry if wrong...

UIAlertAction handler is an optional closure.
Optional closures are implicitly escaped.

Shadowed self has been captured again by the UIAlertAction initializer.
Maybe...

I checked in this way...
( Instruments Leaks is not working in my environment😂)
```swift
extension UIViewController {
    public func alert(title: String?,
                      message: String? = nil,
                      actions: [AlertAction] = [AlertAction(title: "OK")],
                      preferredStyle: UIAlertController.Style = .alert,
                      vc: UIViewController? = nil) -> Observable<OutputAction> {
        let parentVc = vc ?? self
        let alertController = UIAlertController(title: title, message: message, preferredStyle: preferredStyle)
        
        // 👇👇👇👇
        _ = Observable<Int>.interval(.seconds(1), scheduler: MainScheduler.instance)
            .bind(onNext: { [weak alertController] _ in
                print(alertController == nil)
            })
        // ☝️☝️☝️☝️
        
        return
            alertController
                .addAction(actions: actions)
                .do(onSubscribed: {
                    parentVc.present(alertController, animated: true)
                })
    }
}
```

Comment (keisukeYamagishi) [reply RxAlert/UIViewController+.swift]: ViewController where RxAlert was called was deinit was called . So I thought it was good sir.
fix it!

but i use `[unowned self]`

Comment (susan335) [reply RxAlert/UIViewController+.swift]: Thank you!

> but i use [unowned self]

I think it is good too!

**Pull request merged**